pkg/bundle/plain/v0/convert.go

```diff
@@ -0,0 +1,358 @@
+package v0
+
+import (
+	"fmt"
+	"hash/fnv"
+	"strings"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/rukpak/internal/util"
+	"github.com/operator-framework/rukpak/pkg/bundle"
+	registryv1 "github.com/operator-framework/rukpak/pkg/bundle/registry/v1"
+)
+
+// WithInstallNamespace is an option for converting other bundles to `plain+v0`.
+//
+// The option returned by this function can be used in any conversion function that outputs a `plain+v0` bundle. The
+// install namespace is the name of the Kubernetes namespace that the contents of the bundle should be installed to. It
+// can sometimes be inferred by the contents of the source bundle, but not in every case. This option will override any
+// inferred namespace.
+func WithInstallNamespace(namespace string) func(*Bundle) {
+	return func(b *Bundle) {
+		b.installNamespace = namespace
+	}
+}
+
+// WithTargetNamespaces is an option for converting other bundles to `plain+v0`.
+//
+// The option returned by this function can be used in any conversion function that outputs a `plain+v0` bundle. The
+// target namespaces is a list of Kubernetes namespaces that the the bundle's resulting objects expect access to.
+// Depending on the install mode, these namespaces are used when constructing RBAC manifests.
+func WithTargetNamespaces(namespaces ...string) func(*Bundle) {
+	return func(b *Bundle) {
+		b.targetNamespaces = namespaces
+	}
+}
+
+func init() {
+	bundle.RegisterConversionFunc(convertFromRegistryV1)
+}
+
+func convertFromRegistryV1(in registryv1.Bundle, opts ...func(*Bundle)) (*Bundle, error) {
+	b := Bundle{FS: in.FS, createdSvcAccs: make(map[string]struct{})}
+	for _, opt := range opts {
+		opt(&b)
+	}
+	csv, err := in.CSV()
+	if err != nil {
+		return nil, err
+	}
+	if err := b.validateOptions(csv); err != nil {
+		return nil, err
+	}
+	if err := b.extractCsvDeployments(csv); err != nil {
+		return nil, err
+	}
+	if err := b.extractCsvRBAC(csv); err != nil {
+		return nil, err
+	}
+
+	return &b, nil
+}
+
+func (b *Bundle) validateOptions(csv *operatorsv1alpha1.ClusterServiceVersion) error {
+	b.validateInstallNamespace(csv)
+	return b.validateTargetNamespaces(csv)
+}
+
+func (b *Bundle) validateInstallNamespace(csv *operatorsv1alpha1.ClusterServiceVersion) {
+	if b.installNamespace != "" {
+		return
+	}
+	if ns, ok := csv.Annotations["operatorframework.io/suggested-namespace"]; ok {
+		b.installNamespace = ns
+		return
+	}
+
+	// TODO(ryantking): Does srcBundle.CSV().GetName() give us package name
+	// per @joelanford's install namespace detection suggestion?
+	b.installNamespace = fmt.Sprintf("%s-system", csv.GetName())
```

Comment:
No, `csv.GetName()` will return the CSV name (typically something like `foo.v0.1.0`). We want this to be the package name (typically something like `foo`).

The package name can be discovered from a registry+v1 bundle's metadata/annotations.yaml file.